nrf_security/cmake/symbol_rename.cmake

```diff
@@ -12,50 +12,108 @@
+    remove_objects("MBEDTLS_CMAC_C"       ${BACKEND_NAME} remove_line "cmac_alt.c.obj")
+    remove_objects("MBEDTLS_ECP_C"        ${BACKEND_NAME} remove_line "ecp.c.obj")
+    remove_objects("MBEDTLS_ECP_C"        ${BACKEND_NAME} remove_line "ecp_common.c.obj")
+    remove_objects("MBEDTLS_ECP_C"        ${BACKEND_NAME} remove_line "ecp_curves.c.obj")
+    remove_objects("MBEDTLS_ECP_C"        ${BACKEND_NAME} remove_line "ecp_alt_cc310.c.obj")
+    remove_objects("MBEDTLS_ECP_C"        ${BACKEND_NAME} remove_line "ecp_curved_alt.c.obj")
+    remove_objects("MBEDTLS_ECJPAKE_C"    ${BACKEND_NAME} remove_line "ecjpake.c.obj")
+    remove_objects("MBEDTLS_CHACHA20_C"   ${BACKEND_NAME} remove_line "chacha20.c.obj")
+    remove_objects("MBEDTLS_CHACHA20_C"   ${BACKEND_NAME} remove_line "chacha20_alt.c.obj")
+    remove_objects("MBEDTLS_CHACHAPOLY_C" ${BACKEND_NAME} remove_line "chachapoly.c.obj")
+    remove_objects("MBEDTLS_CHACHAPOLY_C" ${BACKEND_NAME} remove_line "chachapoly_alt.c.obj")
+    remove_objects("MBEDTLS_POLY1305_C"   ${BACKEND_NAME} remove_line "poly1305_alt.c.obj")
+    remove_objects("MBEDTLS_POLY1305_C"   ${BACKEND_NAME} remove_line "poly1305.c.obj")
+    remove_objects("MBEDTLS_POLY1305_C"   ${BACKEND_NAME} remove_line "poly.c.obj")
+    remove_objects("MBEDTLS_SHA1_C"       ${BACKEND_NAME} remove_line "sha1.c.obj")
+    remove_objects("MBEDTLS_SHA1_C"       ${BACKEND_NAME} remove_line "sha1_alt.c.obj")
+    remove_objects("MBEDTLS_SHA256_C"     ${BACKEND_NAME} remote_line "sha256.c.obj")
+    remove_objects("MBEDTLS_SHA256_C"     ${BACKEND_NAME} remove_line "sha256_alt.c.obj")
+  endif()
+
   string(TOLOWER "${backend}" MBEDTLS_BACKEND_PREFIX)
-  configure_file(${template}
+  configure_file(${rename_template}
                  symbol_rename_${MBEDTLS_BACKEND_PREFIX}.txt)
-  
+
+  set(redefine_line "--redefine-syms;${CMAKE_CURRENT_BINARY_DIR}/symbol_rename_${MBEDTLS_BACKEND_PREFIX}.txt")
+
   set(BACKEND_RENAMED_LIBARY lib${IMAGE}mbedcrypto_${MBEDTLS_BACKEND_PREFIX}_backend.a)
+  set(temp_target
+    ${CMAKE_CURRENT_BINARY_DIR}/lib${IMAGE}mbedcrypto_${MBEDTLS_BACKEND_PREFIX}_temp.a
+  )
+
+  set(copy_file_command ${CMAKE_COMMAND} -E copy $<TARGET_FILE:${IMAGE}mbedcrypto_${MBEDTLS_BACKEND_PREFIX}> ${temp_target})
+  set(remove_file_command ${CMAKE_COMMAND} -E remove ${temp_target})
+
+  if (${is_stripped})
+    set(remove_object_command ${CMAKE_AR} d ${temp_target} ${remove_line})
+  endif()
+
   add_custom_command(
-    OUTPUT ${BACKEND_RENAMED_LIBARY}
-    COMMAND ${CMAKE_OBJCOPY}
-            --redefine-syms
-            ${CMAKE_CURRENT_BINARY_DIR}/symbol_rename_${MBEDTLS_BACKEND_PREFIX}.txt
-            $<TARGET_FILE:${IMAGE}mbedcrypto_${MBEDTLS_BACKEND_PREFIX}>
+    OUTPUT  ${BACKEND_RENAMED_LIBARY}
+    COMMAND ${copy_file_command}
+    COMMAND ${remove_object_command}
+    COMMAND ${CMAKE_OBJCOPY} ${redefine_line}
+            ${temp_target}
             ${BACKEND_RENAMED_LIBARY}
+    COMMAND ${remove_file_command}
```

Comment:
If you first redefine and then remove the unwanted objects you can skip the creation and removal of the temp file.

Reply:
We wanted to strip before redefining to be "cleaner". We can try the other way around and see if it gives us any problems...

Reply:
Fixed